### PR TITLE
Fixed #24275 -- Used autopep8 to format migration files

### PR DESCRIPTION
**Pull request by darkryder**
After discussion on https://github.com/django/django/pull/4064

### CODE REVIEW

File: django/db/migrations/writer.py [hunk 1]
```diff
@@ -184,7 +184,15 @@ def as_string(self):
...
 
-        return (MIGRATION_TEMPLATE % items).encode("utf8")
+        prepared_migration_statement = (MIGRATION_TEMPLATE % items).encode("utf8")
+        try:
+            import autopep8
+            return autopep8.fix_code(prepared_migration_statement,
+                options=autopep8.parse_args(['--aggressive', '']))
+        except ImportError:
+            pass
+        finally:
```
Comment (coldmind) [inline django/db/migrations/writer.py]: finally will be executed anyway, was there ImportError or not, so autopep8 result will be not returned

Comment (darkryder) [inline django/db/migrations/writer.py]: I don't understand why. I'm returning the fixed_code within the try block itself.

Comment (collinanderson) [inline django/db/migrations/writer.py]: Interesting. I just tried it. This returns 2.

    def func():
        try:
            return 1
        finally:
            return 2

Comment (collinanderson) [inline django/db/migrations/writer.py]: basically, if you out the `return prepared_migration_statement` in the `except:` clause it should work fine.

Comment (aaugustin) [inline django/db/migrations/writer.py]: Anyway, good practices include:

- keeping the try clause as small as possible
- avoiding multiple returns

Hence the correct idiom is:

    try:
        import autopep8
    except ImportError:
        pass
    else:
        prepared_migration_statement = autopep8.fix_code(prepared_migration_statement, ...)
    return prepared_migration_statement

File: django/db/migrations/writer.py [hunk 2]
```diff
@@ -184,7 +184,14 @@ def as_string(self):
...
 
-        return (MIGRATION_TEMPLATE % items).encode("utf8")
+        migration_statement = (MIGRATION_TEMPLATE % items).encode("utf8")
+        try:
+            import autopep8
+            migration_statement = autopep8.fix_code(migration_statement,
+                    options=autopep8.parse_args(['--aggressive', '']))
+        except ImportError:
+            pass
+        return migration_statement
```
Comment (coldmind) [inline django/db/migrations/writer.py]: please move this to `except` and return `autopep8` result in `try`

File: django/db/migrations/writer.py (hunk 1 above)
Comment (darkryder) [inline django/db/migrations/writer.py]: Yeah, I wasn't aware of that before :astonished:

File: django/db/migrations/writer.py [hunk 3]
```diff
@@ -184,7 +184,15 @@ def as_string(self):
...
             items['replaces_str'] = "\n    replaces = %s\n" % self.serialize(self.migration.replaces)[0]
 
-        return (MIGRATION_TEMPLATE % items).encode("utf8")
+        migration_statement = (MIGRATION_TEMPLATE % items).encode("utf8")
+        try:
+            import autopep8
+        except ImportError:
+            pass
+        else:
+            migration_statement = autopep8.fix_code(migration_statement,
```
Comment (MarkusH) [inline django/db/migrations/writer.py]: Can you use hanging indents here:

```python
a = b(
    c,
    d=e,
)
```

**Comment by MarkusH** [comment]
Patch lacks docs and tests. You can add autopep8 to the requirements in `tests/requirements/base.txt`

**Comment by timgraham** [comment]
Closing per ticket.

**Pull request closed without merging**